radis/api/kuruczapi.py

```diff
@@ -0,0 +1,564 @@
+
+Author: Racim Menasria
+Date: June 2023
+"""
+
+import io
+import os
+import pkgutil
+from contextlib import closing
+from io import BytesIO
+
+import mendeleev
+import numpy as np
+import pandas as pd
+import requests
+from tqdm import tqdm
+
+from radis.misc.utils import getProjectRoot
+from radis.phys.air import air2vacuum
+
+
+class AdBKurucz:
+    ccgs = 29979245800.0
+    ecgs = 4.80320450e-10
+    mecgs = 9.10938356e-28
+
+    def __init__(self, atom, ionization_state):
+        self.kurucz_url_base = "http://kurucz.harvard.edu/linelists/gfall/gf"
+        self.hdf5_file = None
+        self.data = None
+        self.pfTdat, self.pfdat = self.load_pf_Barklem2016()
+        self.populations = None
+        self.atom = atom
+        self.ionization_state = ionization_state
+        self.atomic_number = getattr(mendeleev, self.atom).atomic_number
+        # Convert atomic_number to element symbol
+        self.element_symbol = mendeleev.element(int(self.atomic_number)).symbol
+
+        # Construct the key
+
+        if ionization_state == "00":
+            self.key = self.element_symbol + "_I"
+        elif ionization_state == "01":
+            self.key = self.element_symbol + "_II"
+        else:
+            self.key = self.element_symbol + "_III"
+
+    def get_url(self, atomic_number, ionization_state):
+        ionization_state = str(ionization_state).zfill(2)
+        return f"http://kurucz.harvard.edu/linelists/gfall/gf{atomic_number}{ionization_state}.all"
+
+    def load_pf_Barklem2016(self):
+        """Load a table of the partition functions for 284 atomic species.
+
+        Returns
+        -------
+        pfTdat:  pd.DataFrame
+            Steps of temperature (K)
+        pfdat:  pd.DataFrame
+            Partition functions for 284 atomic species
+
+        References
+        ----------
+        `Barklem & Collet (2016), Table 8 <https://doi.org/10.1051/0004-6361/201526961>`_
+
+        """
+        pfT_str = "T[K], 1.00000e-05, 1.00000e-04, 1.00000e-03, 1.00000e-02, 1.00000e-01, 1.50000e-01, 2.00000e-01, 3.00000e-01, 5.00000e-01, 7.00000e-01, 1.00000e+00, 1.30000e+00, 1.70000e+00, 2.00000e+00, 3.00000e+00, 5.00000e+00, 7.00000e+00, 1.00000e+01, 1.50000e+01, 2.00000e+01, 3.00000e+01, 5.00000e+01, 7.00000e+01, 1.00000e+02, 1.30000e+02, 1.70000e+02, 2.00000e+02, 2.50000e+02, 3.00000e+02, 5.00000e+02, 7.00000e+02, 1.00000e+03, 1.50000e+03, 2.00000e+03, 3.00000e+03, 4.00000e+03, 5.00000e+03, 6.00000e+03, 7.00000e+03, 8.00000e+03, 9.00000e+03, 1.00000e+04"
+        pfTdat = pd.read_csv(io.StringIO(pfT_str), sep=",")
+        pfTdat = pd.Series(pfTdat.columns[1:]).astype(
+            "float64"
+        )  # Converts the values to float64, skipping the first value
+
+        with open(os.path.join(getProjectRoot(), "db", "kuruczpartfn.txt"), "r") as f:
+            pfdat = pd.read_csv(f, sep="\s+", comment="#", names=pfTdat.index)
+
+        return pfTdat, pfdat
+
+    def load_atomicdata(self):
```

Comment:
What is the purpose of this function? It's not referenced anywhere except in a test to assess the output format